src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -84,6 +84,12 @@ public void ProcessStart_UseShellExecuteTrue_OpenNano_OpensNano()
             }
         }
 
+        [Theory(Skip= "Needs fix for #22299: Should throw"), InlineData(true), InlineData(false)]
```

Comment:
I debugged this test on both osx and linux.

why should we throw when the program that opens the folder appropriately chooses correct application to open the folder?

In osx, the finder pops up when we run this. 
In linux, the folder listing shows up with "links" I believe.

I could change the behavior to what's requested in issue 22299, by throwing if the filename is a directory. 
But before doing so, I just wanted to raise my observation and ask if it's still a problem not throwing when input is a folder?

Reply:
Also, as suggested by @tarekgh I also tested this on windows. when the StartInfo.FileName is a directory, we do successfully open the folder with File Explorer. we don't throw there either.

Reply:
In my opinion, it should not throw if `UseShellExecute` is **true** and "the program that opens the folder, is appropriately choosing correct application for opening a folder".


What https://github.com/dotnet/corefx/issues/22299 is about (if I understood it correctly):
If `UseShellExecute` is **false** and FileName is a directory, I think Process.Start should throw an error.